backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/BPAAttribute.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.model;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.micronaut.core.annotation.Introspected;
+import lombok.*;
+import org.hyperledger.bpa.controller.api.prooftemplates.Attribute;
+import org.hyperledger.bpa.impl.verification.prooftemplates.ValidAttributeConditionOperator;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Introspected
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
+public class BPAAttribute {
```

Comment:
The original idea of the model package was that it only contains entities that directly map to a database table. The bpa.api package supposed to be the place for internal 'models'. So maybe we should rename api to model and model to entities or something. For now we could move all the related models that are not annotated with `@Entity` into a proof template package below the model package. Also the BPA prefix was intended to specify models that are for internal use and not related to Aries functionality. So maybe AttributeModel would make more sense also to distinguish it from the api Attribute. What do you think?

Reply:
BPAAttribute is for internal use. In my understanding, it does not refer to aries. I did not check the bpa.api package yet. I'll have a look, if it makes sense to move it there.